daemon/kmd/wallet/driver/ledger.go

```diff
@@ -307,16 +308,39 @@ func (lw *LedgerWallet) DeleteMultisigAddr(addr crypto.Digest, pw []byte) error
 }
 
 // SignTransaction implements the Wallet interface.
-func (lw *LedgerWallet) SignTransaction(tx transactions.Transaction, pw []byte) ([]byte, error) {
+func (lw *LedgerWallet) SignTransaction(tx transactions.Transaction, pk crypto.PublicKey, pw []byte) ([]byte, error) {
+	pks, err := lw.ListKeys()
+	if err != nil {
+		return nil, err
+	}
+	// Right now the device only supports one key
+	if len(pks) > 1 {
+		return nil, errors.New("LedgerWallet device only supports one key but ListKeys returned more than one")
+	}
+	if (pk != crypto.PublicKey{}) {
+		// A specific key was requested; return an error if it's not the one on the device.
+		if len(pks) < 1 || pks[0] != crypto.Digest(pk) {
```

Comment:
you check `len(pks) < 1` here before dereferencing with  `pks[0]`, but don't do that check before dereferencing below on line 326. can you please be consistent one way or the other?

Reply:
Good catch, thanks!